File: frontend/src/api/stream.ts
import { apiGet } from "./client";

// 与后端 schemas/stream.py 对应。
// play-url 只返回地址，stream status 返回 ZLM 当前是否在线及观看统计。
export interface PlayUrlResponse {
  stream_id: string;
  protocol: string;
  url: string;
}

export interface StreamStatus {
  stream_id: string;
  online: boolean;
  app: string;
  schema_name?: string | null;
  origin_type?: string | null;
  reader_count: number;
  total_reader_count: number;
  tracks: Record<string, unknown>[];
}

export function getPlayUrl(streamId: string) {
  return apiGet<PlayUrlResponse>(`/streams/${streamId}/play-url`);
}

export function getStreamStatus(streamId: string) {
  return apiGet<StreamStatus>(`/streams/${streamId}/status`);
}

File: backend/app/schemas/stream.py
from pydantic import BaseModel, Field


class PlayUrlResponse(BaseModel):
    stream_id: str
    protocol: str
    url: str


class StreamStatus(BaseModel):
    stream_id: str
    online: bool
    app: str
    schema_name: str | None = None
    origin_type: str | None = None
    reader_count: int = 0
    total_reader_count: int = 0
    tracks: list[dict] = Field(default_factory=list)


class StreamStatusResponse(StreamStatus):
    raw: dict = Field(default_factory=dict)


class VideoSourceStatusItem(StreamStatus):
    id: int
    name: str


class VideoSourceStatusSummary(BaseModel):
    total: int
    online: int
    offline: int
    video_sources: list[VideoSourceStatusItem] = Field(default_factory=list)
